spec/download_metadata_from_onesky_spec.rb

```diff
@@ -0,0 +1,70 @@
+require 'spec_helper'
+
+describe Fastlane::Actions::DownloadMetadataFromOneskyAction do
+  describe 'derives a build code from an AppVersion object' do
+    let(:api_key) { 'dummy-api-key' }
+    let(:api_secret) { 'dummy-api-secret' }
+    let(:project_id) { 123_456 }
+    let(:export_content) { File.read(File.join(File.dirname(__FILE__), 'test-data', 'onesky-metadata.json')) }
+
+    before do
+      client = instance_double(Onesky::Client)
+      allow(Onesky::Client).to receive(:new).and_return(client)
+      project = instance_double(Onesky::Project)
+      allow(client).to receive(:project).and_return(project)
+      allow(project).to receive(:export_multilingual).with(
+        source_file_name: 'store-metadata.xml',
+        file_format: 'I18NEXT_MULTILINGUAL_JSON'
+      ).and_return(export_content)
+    end
+
+    def run_test(metadata_files:, expected_content:)
+      in_tmp_dir do |tmp_dir|
+        run_described_fastlane_action(
+          onesky_api_key: api_key,
+          onesky_api_secret: api_secret,
+          onesky_project_id: project_id,
+          source_file_name: 'store-metadata.xml',
+          metadata_files: metadata_files,
+          locales: Fastlane::LocalesMap.default.to_h(:onesky, :app_store)
+        )
+
+        expected_content.each do |path, content|
+          if content.nil?
+            expect(File).not_to exist(File.join(tmp_dir, 'metadata', path.to_s))
+          else
+            expect(File).to exist(File.join(tmp_dir, 'metadata', path.to_s))
+            expect(File.read(File.join(tmp_dir, 'metadata', path.to_s))).to eq(content)
+          end
+        end
+      end
+    end
+
+    it 'works' do
```

Comment:
WIP. We need to add a lot more tests.

Including ones around composite keys (keys which contain `.` in their name, [which OneSky exports as a tree of keys and subkeys split around that `.`](https://github.com/wordpress-mobile/release-toolkit/pull/563/files#diff-332a57bea137650989cb66f84584f9a9bcb128bf42118e3f1c30d9c54e341827R65))